src/badguy/badguy.cpp

```diff
@@ -315,12 +317,18 @@ BadGuy::update(float dt_sec)
     case STATE_SQUISHED:
       m_is_active_flag = false;
       if (m_state_timer.check()) {
-        remove_me();
+        m_alpha_before_fadeout = m_sprite->get_alpha();
+        set_state(STATE_SQUISHED_FADING_OUT);
         break;
       }
       m_col.set_movement(m_physic.get_movement(dt_sec));
       break;
-
+    case STATE_SQUISHED_FADING_OUT:
+      if (m_state_timer.check())
+        remove_me();
+      else
+        m_sprite->set_alpha(m_alpha_before_fadeout - (m_alpha_before_fadeout * m_state_timer.get_progress()));
```

Comment:
And here you'd only need to put m_state_timer.get_progress()
```suggestion
        m_sprite->set_alpha(m_state_timer.get_progress()));
```

Reply:
If alpha was already lower than 1, the suggested change would put it up to 1 before fading out again.

Reply:
Valid point here. Although all badguys currently have alpha=1.0, it doesn't have to hold in future. If they already start at a lower value of alpha, then the suggested change won't work. Just for better readability, I would write it as:
```
m_alpha_before_fadeout * (1.0f - m_state_timer.get_progress())
```